src/OpenTelemetry.Extensions.PersistentStorage.Api/IPersistentBlob.cs

```diff
@@ -0,0 +1,74 @@
+﻿// <copyright file="IPersistentBlob.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Extensions.PersistentStorage.Api
+{
+    /// <summary>
+    /// Represents a persistent blob.
+    /// </summary>
+    public interface IPersistentBlob
+    {
+        /// <summary>
+        /// Reads the content from the blob.
+        /// </summary>
+        /// <returns>
+        /// The content of the blob if the operation succeeded, otherwise null.
+        /// </returns>
+        /// <remarks>
+        /// This function should never throw exception.
+        /// </remarks>
+        byte[] Read();
```

Comment:
Will reading/writing blobs always be synchronous? Any thoughts on supporting async reads/writes in the future?

Reply:
+1

Reply:
Yes, can be added in future. I'm imagining an async call may be desirable if reading/writing blobs over the network - maybe using HttpClient or a messaging client.

Being able to control a timeout could also be nice - either by passing a timeout interval or a cancellation token to Read/Write.